src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonSchemaCreateContext.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json.Schema;
+using System.Text.Json.Serialization.Metadata;
+
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Defines the context in which a JSON schema within a type graph is being generated.
+/// </summary>
```

Comment:
How is someone supposed to create one of these? There's no public ctor, and it looks like using `default(AIJsonSchemaCreateContext)` will blow up if any of its members are used. Presumably then this is only supposed to be created by this component, in order to be passed into the delegate? Consider mentioning that here.